imap/annotate.c

```diff
@@ -1204,19 +1202,7 @@ static int annotatemore_findall_full(const char *pattern, /* internal */
     frock.since_modseq = since_modseq;
     frock.flags = flags;
 
-    /* special case where mboxname is "" and others are empty, this means
-     * server annotations, which are recognised by an empty pattern by the
-     * code below */
-    if (!mailbox && !pattern && !*mboxname) {
-        mboxname = NULL;
-        pattern = "";
-    }
-
-    if (mailbox || mboxname) {
-        if (mailbox) mboxname = mailbox_name(mailbox);
-        r = mboxlist_lookup_allow_all(mboxname, &mbentry, NULL);
-        if (r) goto out;
-    }
+    if (mailbox) mbentry = mailbox_mbentry(mailbox);
```

Comment:
Should we lift this into all the callers as well?

Reply:
I think I had in annotatemore _findall_mailbox() and moved it out.  I don't recall why and it probably wasn't a good reason.  It probably should be there.